eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -145,12 +153,13 @@ object Validation {
 
       publicChannel_opt match {
         case Some(pc) =>
-          // note: if the channel is graduating from private to public, the implementation (in the LocalChannelUpdate handler) guarantees that we will process a new channel_update
-          // right after the channel_announcement, channel_updates will be moved from private to public at that time
           val d1 = d0.copy(
             channels = d0.channels + (c.shortChannelId -> pc),
             privateChannels = d0.privateChannels - c.shortChannelId, // we remove fake announcements that we may have made before
-            rebroadcast = d0.rebroadcast.copy(channels = d0.rebroadcast.channels + (c -> d0.awaiting.getOrElse(c, Nil).toSet)), // we also add the newly validated channels to the rebroadcast queue
+            rebroadcast = d0.rebroadcast.copy(
+              channels = d0.rebroadcast.channels + (c -> d0.awaiting.getOrElse(c, Nil).toSet), // we rebroadcast the channel to our peers
+              updates = d0.rebroadcast.updates ++ (pc.update_1_opt.toSet ++ pc.update_2_opt.toSet).map(u => u -> (if (pc.getNodeIdSameSideAs(u) == nodeParams.nodeId) Set[GossipOrigin](LocalGossip) else Set.empty[GossipOrigin])).toMap // those updates are only defined if this was a previously an unannounced local channel, we broadcast them
+            ), // we also add the newly validated channels to the rebroadcast queue
```

Comment:
I think it's worth extracting the updates to a separate line for readability:

```scala
// those updates are only defined if this was a previously an unannounced local channel, we broadcast them
val previousUpdates = (pc.update_1_opt.toSet ++ pc.update_2_opt.toSet).map(u => u -> (if (pc.getNodeIdSameSideAs(u) == nodeParams.nodeId) Set[GossipOrigin](LocalGossip) else Set.empty[GossipOrigin])).toMap
```

```suggestion
            rebroadcast = d0.rebroadcast.copy(
              channels = d0.rebroadcast.channels + (c -> d0.awaiting.getOrElse(c, Nil).toSet), // we rebroadcast the channel to our peers
              updates = d0.rebroadcast.updates ++ localUpdates
            ), // we also add the newly validated channels to the rebroadcast queue
```

Looking ahead, isn't there a risk here that we'll broadcast an update that contains an alias `short_channel_id` (which doesn't make sense, others won't be able to validate it)? Shouldn't we wait to receive a new `channel_update` with the final `short_channel_id`?

Reply:
> Looking ahead, isn't there a risk here that we'll broadcast an update that contains an alias short_channel_id (which doesn't make sense, others won't be able to validate it)? Shouldn't we wait to receive a new channel_update with the final short_channel_id?

Nice catch. Either we:
(a) only add `channel_update` to the `PublicChannel` if they use the real scid
(b) only broadcast if the `channel_update` uses the real `scid`

I'm tempted to go with (b), so we don't risk losing information, but it feels like (a) is more correct. Otherwise we will definitely send out `channel_update` with aliases in the context of channel queries.

Reply:
I'm also more tempted to go with (b), because it lets us keep creating invoices with routing hints even before we receive the "official" channel update with the real scid. Also, keeping channel updates with aliases around isn't an issue as long as it stays internal to eclair, it only becomes an issue if we broadcast them, so it make sense to have filtering done in the broadcast code, doesn't it?

Reply:
> I think it's worth extracting the updates to a separate line for readability:

Done in a7a35c3bb.

WDYT of the following?

```scala
val dx = d0
  .modify(_.channels).using(_ + (c.shortChannelId -> pc))
  .modify(_.privateChannels).using(_ - c.shortChannelId) // we remove fake announcements that we may have made before
  .modify(_.rebroadcast.channels).using(_ + (c -> d0.awaiting.getOrElse(c, Nil).toSet)) // we rebroadcast the channel to our peers)
  .modify(_.rebroadcast.updates).using(_ ++ updates1)
  .modify(_.stash).setTo(stash1)
  .modify(_.awaiting).setTo(awaiting1)
```